homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -465,41 +476,54 @@ def color_mode(self):
             return ColorMode.HS
         return ColorMode.COLOR_TEMP
 
-    def _set_flash_and_transition(self, message, **kwargs):
+    def _set_flash_and_transition(self, message: dict[str, Any], **kwargs: Any) -> None:
         if ATTR_TRANSITION in kwargs:
             message["transition"] = kwargs[ATTR_TRANSITION]
 
         if ATTR_FLASH in kwargs:
-            flash = kwargs.get(ATTR_FLASH)
+            flash: Any = kwargs.get(ATTR_FLASH)
 
             if flash == FLASH_LONG:
                 message["flash"] = self._flash_times[CONF_FLASH_TIME_LONG]
             elif flash == FLASH_SHORT:
                 message["flash"] = self._flash_times[CONF_FLASH_TIME_SHORT]
 
-    def _scale_rgbxx(self, rgbxx, kwargs):
+    def _scale_rgbxx(
+        self,
+        rgbxx: tuple[int, ...],
+        kwargs: dict[str, Any],
+    ) -> tuple[int, ...]:
         # If there's a brightness topic set, we don't want to scale the
         # RGBxx values given using the brightness.
+        brightness: int
         if self._config[CONF_BRIGHTNESS]:
             brightness = 255
         else:
             brightness = kwargs.get(ATTR_BRIGHTNESS, 255)
-        return tuple(round(i / 255 * brightness) for i in rgbxx)
+        scaled_rgb = tuple(round(i / 255 * brightness) for i in rgbxx)
+        return scaled_rgb
 
-    def _supports_color_mode(self, color_mode):
+    def _supports_color_mode(self, color_mode: ColorMode | str) -> bool:
         """Return True if the light natively supports a color mode."""
         return (
-            self._config[CONF_COLOR_MODE] and color_mode in self.supported_color_modes
+            self.supported_color_modes is not None
+            and self._config[CONF_COLOR_MODE]
+            and color_mode in self.supported_color_modes
         )
 
-    async def async_turn_on(self, **kwargs):  # noqa: C901
+    async def async_turn_on(self, **kwargs: Any) -> None:  # noqa: C901
         """Turn the device on.
 
         This method is a coroutine.
         """
-        should_update = False
-
-        message = {"state": "ON"}
+        brightness: int
+        should_update: bool = False
+        hs_color: tuple[float, float]
+        message: dict[str, Any] = {"state": "ON"}
+        rgb: tuple[int, int, int]
+        rgbw: tuple[int, int, int, int]
+        rgbcw: tuple[int, int, int, int, int]
```

Comment:
I think you can leave these three to be inferred (`tuple[int, ...]`) and use cast only on `_attr_`
```suggestion
```